### ISSUE TITLE
Overflow in loop

### ISSUE THREAD

**Issue by FranciscoDomingues** [vyperlang/vyper#1712]
### Version Information

Tested on Remix

### What's your issue about?

It appears there is no overflow checking for the loop variables. For example, the code 
```
@public
def test_for() -> int128:
  a: int128 = 0
  for i in range(MAX_INT128, MAX_INT128+2):
        a = i
  return a
```
returns -2**127.

**Comment by fubuloubu** [comment]
Good catch!

**Comment by jacqueswww** [comment]
Looking at this.

**Comment by fubuloubu** [comment]
Fixed in #2029

Error:
```
E           vyper.exceptions.TypeMismatch: Given reference has type uint256, expected int128
E           function 'test_for', line 6:12 
E                5     for i in range(MAX_INT128, MAX_INT128+2):
E           ---> 6         a = i
E           -------------------^
E                7     return a
```

Added this test case in
